src/main/java/net/coderbot/iris/pipeline/DeferredWorldRenderingPipeline.java

```diff
@@ -513,10 +514,19 @@ private Pass createPass(ProgramSource source) {
 		Objects.requireNonNull(source.getVertexSource());
 		Objects.requireNonNull(source.getFragmentSource());
 		ProgramBuilder builder;
+		String vertex = source.getVertexSource().orElse(null);
+		String geometry = source.getGeometrySource().orElse(null);
+		String fragment = source.getFragmentSource().orElse(null);
+
+		if (source.getName().contains("entities")) {
```

Comment:
This is incorrect, entityColor is available to all programs, but it will only have a value not equal to `vec4(0.0)` in entity rendering - which might not use `gbuffers_entities`. This code will need to be able to keep track of whether the overlay vertex attribute is in use, and only use the patched program variant if it is.

Reply:
is this better?